include/bluetooth/att.h

```diff
@@ -54,9 +55,15 @@ extern "C" {
 #define BT_ATT_LAST_ATTRIBUTE_HANDLE            0xffff
 #define BT_ATT_LAST_ATTTRIBUTE_HANDLE __DEPRECATED_MACRO BT_ATT_LAST_ATTRIBUTE_HANDLE
 
+#if defined(CONFIG_BT_EATT)
 #if defined(CONFIG_BT_TESTING)
+
 int bt_eatt_disconnect_one(struct bt_conn *conn);
+int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
+uint16_t bt_att_num_eatt(struct bt_conn *conn);
```

Comment:
Since these are in a public header file, shouldn't they be documented?

Regarding `bt_att_num_eatt()` I think I'd prefer `bt_att_eatt_count()` as the name. Also, why the fixed-width `uint16_t` as the return value? Why not e.g. `size_t`?

Reply:
I agree to all points above, please address them @hermabe.

Reply:
@jhedberg regarding the uint16_t/size_t I had though the same, but if memory serves me correct the PSM is 16-bit, and there can't be any more channels than PSMs, so I actually ended up with the conclusion that uin16_t was a fine choice, as that matches the maximum number of channels. 

Given that these functions are just for testing, perhaps they shouldn't even be in the public header? Is there anything wrong with tests including non-public headers for testing?

Reply:
- No good reason for the return type. Will change to `size_t`.
- I am fine with changing the name to `bt_eatt_count()`
- These are guarded by `CONFIG_BT_TESTING` and currently only meant for internal testing use. I will make them public in a future PR and document them then. Is the documentation needed now? I see the already existing `bt_eatt_disconnect_one()` does not have any documentation.

Reply:
I'm ok skipping the documentation while behind `BT_TESTING`, although it wouldn't hurt to have it (e.g. would have made my review easier)